app/helpers/remote_execution_helper.rb

```diff
@@ -161,7 +161,7 @@ def invocation_result(invocation, key)
 
   def preview_box(template_invocation, target)
     renderer = InputTemplateRenderer.new(template_invocation.template, target, template_invocation)
-    if (preview = renderer.preview)
+    if (preview = (load_template_from_task(template_invocation, target) || renderer.preview))
```

Comment:
what are the conditions for input not yet have rendered template? e.g. future execution?

Reply:
For future execution the task object would not exist at all (only the parent task exists at that time). This is to prevent a race condition where the `RunHostJob` already exists, but its `ProxyAction` was not planned yet

Reply:
I guess I should have attached the benchmarks to the PR